### ISSUE TITLE
update file creates new one

### ISSUE THREAD

**Issue by ricebus** [mxmlnkn/ratarmount#90]
Hi

I'm trying to update a file inside an archive but instead of updating the file it just creates a new one.
I've attached a screenshot and a little script that reproduces the issue

![image](https://user-images.githubusercontent.com/12892036/173240752-f23e6700-7ec6-43f7-94b8-7de0c0d52e1a.png)

```
cd ~
\rm -rf ~/test
mkdir -p test/tmpdir
cd ~/test/tmpdir
echo "hello" > world
cd ~/test
tar cf archive.tar -C tmpdir . --owner 0 --group 0 --numeric-owner
tar tvf archive.tar
mkdir root rootw
~/ratarmount-v0.11.2-manylinux2014_x86_64.AppImage archive.tar root -w rootw
echo "world" >> root/world
echo commit | ~/ratarmount-v0.11.2-manylinux2014_x86_64.AppImage archive.tar root -w rootw --commit-overlay
fusermount -u root
tar tvf archive.tar
```

**Comment by mxmlnkn** [comment]
The problem is caused by the leading `./`. I already delete not only the file name itself but also `/filename` and now I also have to add `./filename` to the deletion list. I hope no tool generates files named like `././filename` or paths nested even deeper. TAR definitely supports writing things like that into the TAR file header.

I pushed a fix in 3bf88fc on the develop branch.

In order to be really robust, I might have to roll out my own tar editing from inside Python. But, Python's tarfile module does not seem to support removing files. I would have to copy-write the whole archive to another file using tarfile while skipping all files to delete. It might be hard to do this in-place with the tarfile module. Also, it feels like this might introduce subtle changes to the TAR archive, e.g., the specific TAR format type (GNU tar, ...).

**Comment by ricebus** [comment]
Will this work with directories starting with "./"?
Also, will the resulting tar act the same although the resulting tar looks a bit different (without the leading./)

Are there appImages of the dev branch for testing?

**Comment by mxmlnkn** [comment]
Yes, it should work with directories starting with `./`. I found two other bugs, though, when trying to delete a directory. I pushed fixes for them. As far as I know, there should be no difference between whether there is a leading `./` or not. Normally, it would only matter for extracting the archive and in that use case, it seems equivalent. But options that read and work on the metadata like the GNU tar `--transform=EXPRESSION` might behave differently. The same would apply to deleting files and others. Those might be relevant if you do automated scripting and modifications on TARs but it seems a rare use case to me.

No, there are no AppImages for the dev branch. I can build them on-demand but I think the pipeline artifacts are protected and I would have to attach them anyway or publish them somewhere. I don't know where to put them. I'm also hesitant to attach 14MB too often.

[ratarmount-manylinux2014_x86_64.AppImage.zip](https://github.com/mxmlnkn/ratarmount/files/8886565/ratarmount-manylinux2014_x86_64.AppImage.zip)

You already built AppImages in your fork before, didn't you?
The easiest for me would be if you could update the develop branch in your fork to mine and then trigger the AppImage pipeline manually. You can even cancel the AppImage-tests, which run after the AppImage has been built in that pipeline, and the artifact will still be downloadable. I tried that just now.

**Comment by ricebus** [comment]
This one just spams my output:
/bin/bash: /tmp/.mount_ratarmW2dGhP/usr/lib/libtinfo.so.5: no version information available (required by /bin/bash)
And no, I didn't fork this. I got here from some stackoverflow thread where you suggested your tool :)

**Comment by mxmlnkn** [comment]
It's interesting to hear how people found out about ratarmount :).

As for the AppImage, I'm so sorry for the trouble. It seems like the one time, I skip over the tests, something went wrong. I can reproduce problems. For me it seems to recursively call itself and then quits with the error: `/bin/bash: warning: shell level (1000) too high, resetting to 1`. It seems to be recursively opening itself. I didn't even change anything in that regard since the last working AppImage. And, simply rebuilding the AppImage already fixes the problem. Maybe something went wrong upstream somewhere for one of the dependencies I'm using.

Here is an AppImage that works for me:

[ratarmount-manylinux2014_x86_64.AppImage.zip](https://github.com/mxmlnkn/ratarmount/files/8893321/ratarmount-manylinux2014_x86_64.AppImage.zip)

**Comment by mxmlnkn** [comment]
Should be fixed in 0.11.3.
